blocks/sharepoint/sharepoint.ts

```diff
@@ -256,7 +282,7 @@ export class Sharepoint {
             const msgraphClient = cbk.apiClient.msgraph;
 
             let uploadSessionURL;
-            if (folderId) {
+            if (folderId && !isNaN(Number(folderId))) {
```

Comment:
@dylanoldfield Could you please explain why we are checking for isNaN here?
And it would be helpful to create a function with a descriptive name to clarify its purpose for future reference.

Reply:
Because in the selectInput drop down, the associated value for the list items its the folderId which is a number. However we want to support {{Vars}} which do not have an underlying id, so that I why I check for NaN